docs/NNNN-mutable-head.md

```diff
@@ -0,0 +1,439 @@
+a version conflict, then it is up to the client implementation to decide what to
+do.
+
+## Implementation Notes
+
+Note: The following notes reference content paths as `[version]/content` and the
+inventory sidecar as `inventory.json.sha512`. These are the default OCFL values,
+but they can differ based on inventory file settings.
+
+### Mutable HEAD Creation
+
+When an object does not have an active mutable HEAD, then a new mutable version
+is created following similar steps as outlined in the [OCFL implementation
+notes](https://ocfl.io/draft/implementation-notes/#an-example-approach-to-updating-ocfl-object-versions)
+with two notable differences.
+
+1. The content paths of newly added files must be relative to
+   `extensions/NNNN-mutable-head/head/content/r1`.
+2. Instead of moving the new version to `[object-root]/vN`, it must be moved to
+   `[object-root]/extensions/NNNN-mutable-head/head`.
+
+The procedure for creating a new mutable HEAD is as follows:
+
+1. Stage a new OCFL version as you would normally, noting the above exceptions.
+2. Write the file `[object-root]/extensions/NNNN-mutable-head/revisions/r1`
+   containing the text `r1`. If this fails, abort because another process
+   already created a mutable HEAD.
+3. Move the staged OCFL version to `[object-root]/extensions/NNNN-mutable-head/head`.
+4. Copy `[object-root]/inventory.json.sha512` to `[object-root]/extensions/NNNN-mutable-head/root-inventory.json.sha512`
+
+### Mutable HEAD Mutation
+
+Mutating an existing mutable HEAD requires creating a new revision and is more
+involved than the standard process for creating a new OCFL version. The
+following process should provide close to the same level of safety as is
+expected when creating a new OCFL version.
+
+1. Identify the next available revision number, `rN`, by inspecting the revision
+   markers in `[object-root]/extensions/NNNN-mutable-head/revisions`.
+2. Create a new revision directory somewhere in temp space named `rN`.
+3. Create a copy of the existing mutable HEAD inventory, hence forth referred to
+   as the "new inventory".
+4. Write any files that are added with digests not in the mutable HEAD
+   inventory's manifest to the temp revision directory, and add corresponding
+   manifest entries to the new inventory.
+5. Update the version state in the new inventory to reflect any changes that
+   were made to the object (deletions, renames, additions, modifications).
+6. Update the new inventory’s manifest and fixity blocks, removing any entries
+   that were introduced in prior revisions of the mutable HEAD but are no longer
+   referenced in the current version state.
+7. Write the new inventory and its sidecar to temp space.
+8. Write the file `[object-root]/extensions/NNNN-mutable-head/revisions/rN`
+   containing the text `rN`. If this fails, abort because another process
+   already created a revision using the same revision number.
+9. Move the temp revision directory to
```

Comment:
Should this say, "If the temp revision directory is not empty, move it to..." because of the rule in the Content Directory that _If a revision does not add any new files, then a new content revision directory MUST NOT be created._?

Also, should "temp" be changed to "temporary" everywhere?